src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs

```diff
@@ -125,6 +128,7 @@ private DependencyContext LoadAssemblyContext(Assembly assembly, IDependencyCont
             return null;
         }
 
+        [RequiresAssemblyFiles(Message = "Needs additional support for reading .deps.json in single file", Url = "https://github.com/dotnet/runtime/issues/41265")]
```

Comment:
This should be reverted. If you look at the code of this method, it is 100% safe and doesn't "RequiresAssemblyFiles".

```C#
            // Assemblies loaded in memory (e.g. single file) return empty string from Location.
            // In these cases, don't try probing next to the assembly.
            string assemblyLocation = assembly.Location;
            if (string.IsNullOrEmpty(assemblyLocation))
            {
                return null;
            }
```

Reply:
After https://github.com/dotnet/runtime/pull/41430 was merged we don't crash but we don't have a way to actually read the content of the application's .deps.json file which is currently not accessible in any way. So I think this pattern is still single file incompatible, I guess @vitek-karas  has more insight on this

Reply:
I think we should keep the warning here - it's true that there will be callers which handle null return value and those would just suppress the warning, but most callers I've seen don't expect null, so in this specific case I think it makes sense to keep the warning. Also because it basically says - calling this for single-file is a no-op, the feature just doesn't work at all in single-file.